From: Michael Casadevall <mcasadevall@debian.org>
Subject: Respect Qt SIP flags
Forwarded: no
Origin:
Last-Update: 2009-04-15

Corrects FTBFS on armel by making sure pykde4 properly parses the python-qt4
flags passed from PythonQt4's cmake file.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -103,6 +103,9 @@ include_directories(
     ${KDEPIMLIBS_INCLUDE_DIRS}
 )
 
+# Pull in ARM configuration options if needed
+STRING(REGEX MATCH "PyQt_qreal_double" SIP_ARM_HACK ${PYQT4_SIP_FLAGS} "")
+
 set(SIP_INCLUDES ${PYQT4_SIP_DIR} sip)
 set(SIP_CONCAT_PARTS 8)
 if (WIN32)
@@ -110,7 +113,7 @@ if (WIN32)
 else ()
     set(SIP_TAGS ALL WS_X11 ${PYQT4_VERSION_TAG})
 endif ()
-set(SIP_DISABLE_FEATURES VendorID PyQt_NoPrintRangeBug)
+set(SIP_DISABLE_FEATURES VendorID PyQt_NoPrintRangeBug ${SIP_ARM_HACK})
 
 # Use an extra option when compiling on Python 3.
 if (PYTHON_VERSION_MAJOR GREATER 2)
